File: algorithms/easy/two_sum/twosum.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func twoSum(nums []int, target int) []int {
	var result []int
	for i := 0; i < len(nums)-1; i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[i]+nums[j] == target {
				result = append(result, i, j)
				break
			}
		}

		if len(result) == 2 {
			break
		}
	}

	return result
}

func main() {
	file, err := os.Open("two_sum.txt")
	if err != nil {
		log.Fatal(err.Error())
	}
	defer file.Close()

	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	for i := 0; i < len(lines)-1; i += 2 {
		a := strings.Trim(lines[i], "[]")
		b := strings.Split(a, ", ")
		nums := make([]int, len(b))
		for j, k := range b {
			nums[j], _ = strconv.Atoi(k)
		}
		target, _ := strconv.Atoi(lines[i+1])
		fmt.Println(twoSum(nums, target))
	}
}

File: algorithms/easy/two_sum/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        
        for i in range(0, len(nums)-1):
            for j in range(i+1, len(nums)):
                if(nums[i] + nums[j] == target):
                    result.append(i)
                    result.append(j)
                    break
            
            if(len(result) == 2) :
                break
                
        return result

f = open("two_sum.txt", "r+")
r = f.read().splitlines()
for i in range(0, len(r)-1, 2):
    nums = r[i]
    target = r[i+1]
    solution = Solution()
    result = solution.twoSum(eval(nums), int(target))
    print(result)